sweepbatcher/presigned.go

```diff
@@ -0,0 +1,457 @@
+package sweepbatcher
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/btcsuite/btcd/blockchain"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwallet/chainfee"
+)
+
+// ensurePresigned checks that we can sign a 1:1 transaction sweeping the input.
+func (b *batch) ensurePresigned(ctx context.Context, newSweeps []*sweep) error {
```

Comment:
does `ensurePresigned` imply that there should be already be a transaction with given inputs that has a populated witness? I don't see a check anywhere in this method or sub-method that checks for a witness.

Reply:
I added a check to `CheckSignedTx` which is called by this method:

```go
// Make sure all inputs of signedTx have a non-empty witness.
for _, txIn := range signedTx.TxIn {
	if len(txIn.Witness) == 0 {
		return fmt.Errorf("input %s of signed tx is not signed",
			txIn.PreviousOutPoint)
	}
}
```